tools/benchmarksAnalyzer/src/main/kotlin/org/jetbrains/analyzer/SummaryBenchmarksReport.kt

```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.jetbrains.benchmarksAnalyzer
+import org.jetbrains.report.BenchmarkResult
+import org.jetbrains.report.Environment
+import org.jetbrains.report.Compiler
+import org.jetbrains.report.BenchmarksReport
+import kotlin.math.abs
+
+typealias SummaryBench = Pair<MeanVarianceBench?, MeanVarianceBench?>
+typealias BenchTable = Map<String, MeanVarianceBench>
+typealias SummaryBenchTable = Map<String, SummaryBench>
+typealias ScoreChange = Pair<MeanVariance, MeanVariance>
+
+// Summary report with comparasion of separate benchmarks results
+class SummaryBenchmarksReport (val currentReport: BenchmarksReport,
+                               val previousReport: BenchmarksReport? = null,
+                               val meaningfulChangesValue: Double = 0.5) {
+
+    // Lists of benchmarks in different status
+    private val failedBenchs = mutableListOf<String>()
```

Comment:
Why not lateinit immutable lists?

Reply:
I changed where it is possible, but somewhere buckets are intersected. Separate filling of them added extra looping over results.